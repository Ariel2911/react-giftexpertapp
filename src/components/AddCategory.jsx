import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
    // console.log('handleInputValue llamado');
  };

  const handleSubmit = (e) => {
    console.log('handleSubmit llamado');
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={(e) => handleInputValue(e)}
      />
      <button>Add</button>
    </form>
  );
};

AddCategory.protoType = {
  setCategories: PropTypes.func.isRequired,
};
