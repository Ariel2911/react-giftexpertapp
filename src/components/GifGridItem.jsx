export const GifGridItem = ({ id, title, url }) => {
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        border: '1px solid black',
        borderRadius: '8px',
        margin: '4px',
        width: '150px',
        height: '160px',
      }}
    >
      <img src={url} alt={title} style={{ width: '100%', height: '75%' }} />
      <p
        style={{
          textAlign: 'center',
          height: '25%',
          margin: '0',
          overflow: 'hidden',
        }}
      >
        {title}
      </p>
    </div>
  );
};
