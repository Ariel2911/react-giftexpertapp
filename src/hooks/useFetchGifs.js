import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/GetGifs';

export const useFetchGifs = (category) => {
  const [images, setImages] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((gifs) => {
      setImages({
        data: gifs,
        loading: false,
      });
    });
  }, [category]);

  // console.log(images)
  return images;
};
