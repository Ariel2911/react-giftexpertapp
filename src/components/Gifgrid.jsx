import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <li style={{ display: 'flex', flexWrap: 'wrap' }}>
        {loading ? (
          <p>Cargando</p>
        ) : (
          images.map((image) => {
            return <GifGridItem key={image.id} {...image} />;
          })
        )}
      </li>
    </>
  );
};
