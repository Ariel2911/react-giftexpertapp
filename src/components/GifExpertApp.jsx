import { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './Gifgrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState( defaultCategories );

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ul>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ul>
    </>
  );
};
