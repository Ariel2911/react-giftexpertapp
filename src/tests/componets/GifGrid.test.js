import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/Gifgrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas de <GifGrid />', () => {

    const category = 'cat';
    
    test('Debe de mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category} />)
        expect( wrapper ).toMatchSnapshot();
        
    });
    
    test('Debe de mostrar items cuando se cargan  imagenes useFetchGifs', () => {

        const gifs =[
            {
            id: 'abc',
            url:'https:/local/cualquiercosa.jpg',
            title: 'Cualquier Cosa'
            },            
            {
            id: 'abd',
            url:'https:/local/cualquiercosa.jpg',
            title: 'Cualquier Cosa'
            },

        ];
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    })
});