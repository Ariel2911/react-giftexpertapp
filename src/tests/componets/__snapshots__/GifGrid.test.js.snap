// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid /> Debe de mostrar items cuando se cargan  imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    cat
  </h3>
  <li
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
      }
    }
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier Cosa"
      url="https:/local/cualquiercosa.jpg"
    />
    <GifGridItem
      id="abd"
      key="abd"
      title="Cualquier Cosa"
      url="https:/local/cualquiercosa.jpg"
    />
  </li>
</Fragment>
`;

exports[`Pruebas de <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    cat
  </h3>
  <li
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
      }
    }
  >
    <p>
      Cargando
    </p>
  </li>
</Fragment>
`;
